src/figuresClasses.ts

```diff
@@ -1,11 +1,79 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('The side must be greater than 0');
+    }
+
+    if (Math.max(a, b, c) >= a + b + c - Math.max(a, b, c)) {
+      throw new Error(
+        'Longest side triangle cannot be greater than the sum of the other two',
+      );
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+    const s: number = (a + b + c) / 2;
+    const area: number = Math.sqrt(s * (s - a) * (s - b) * (s - c));
+
+    return Math.floor(area * 100) / 100;
```

Comment:
[CODE STYLE] - The `getArea` method should not accept any parameters. All necessary values are already available as object fields. Please remove the destructuring in line 31 as it is unnecessary.